### DISCUSSION TITLE
PlexCleaner removing forced subtitles

### DISCUSSION THREAD

**Discussion by C0nw0nk** [ptr727/PlexCleaner#167] [General]
So I've been using plexcleaner for months now my library is perfect thanks to @ptr727 

Direct playback works great on all media the only issue is media with forced subtitles will cause plex to start transcoding.

The solution is to remove all forced subtitles since I dont need them. I am keeping English subtitles just removing any that are tagged as forced.

But the question is the configuration or command to do this on a full library.

I was looking at `removesubtitles`

**Comment by ptr727** [comment]
Forced subtitles is just a flag on the track, you could experiment with mkvpropedit or mkvtoolnix GUI tools to remove the forced flag, and see if transcoding still happens.

My guess is it is not the forced flag, but the type of subtitle, where some players cannot burn the subtitles at the player side, the server has to burn them in the stream sent to the player.

The `removesubtitles` command is a nuclear option that removes all subtitle tracks.

**Reply by C0nw0nk** [reply]
Yeah that's the dilemma I do not want to remove all subtitles because i wanted to keep English just remove the forced flags.

I will have to figure out a way to scan through every single media file and list all those with forced subtitles so i can then use that list to remove the forced subtitles only.

**Comment by C0nw0nk** [comment]
# mkvpropedit
`C:\Users\Con\Desktop\plexcleaner\win-x64\Tools\MkvToolNix\mkvpropedit.exe "C:\Users\Con\Desktop\test.mkv" --edit track:s1 --set flag-forced=0`
`C:\Users\Con\Desktop\plexcleaner\win-x64\Tools\MkvToolNix\mkvpropedit.exe "C:\Users\Con\Desktop\test.mkv" --edit track:s1 --set flag-default=0`

```
The file is being analyzed.
The changes are written to the file.
Done.
```

**Reply by C0nw0nk** [reply]
@ptr727 This will remove the default and the forced flags.

The question now is how can i use this in plexcleaner to go through the entire library ?

It would also need to apply to all found subtitle tracks `--edit track:s1` only applies to the first. if theres 10 subs it will need to apply for all s1,s2,s3,s4,s5,s6 etc

**Reply by ptr727** [reply]
There is no such ability in the tool.  
You can do it using e.g. a [script](https://stackoverflow.com/questions/54608762/batch-changing-titles-of-mkv-files-in-multiple-directories)

**Comment by C0nw0nk** [comment]
@ptr727 

Did it and got it working here is my script if anyone wants to use it on their plex folders.

It will recursively go through folders and subfolders of network shares or drive paths.

Made it count through all subtitles and ensure every single subtitle inside the file has its default and forced flags removed that way i get to keep the subtitles in-case a user wants to manually enable subs.

https://github.com/C0nw0nk/DisableForcedSubtitles

**Comment by C0nw0nk** [comment]
@ptr727 I have updated it.

https://github.com/ptr727/PlexCleaner/discussions/167#discussioncomment-6153335

The way i use this now is like this.

1. New Media Content added to plex library
2. Run `mkvremovesubsflags.cmd` since newly added files will not have `.PlexCleaner` sidecar files allowing me to disable forced subtitles
3. Run PlexCleaner after so the new content gets converted for direct playback inline with settings.json
4. Run in loop so new media added gets flags removed then converted after.

This specific order to prevent modifying already modified files obviously.
```
:: Check for the .PlexCleaner sidecar file to skip already modified media
:: 1 is to skip
:: 0 is to modify all
set check_for_sidecar=1
```

**Comment by lucas-0liveira** [comment]
The solution I found is to remove all forced subtitles since I don't need them. I'm keeping the English subtitles and removing the ones marked as forced.

But the question is about the setup or command to do this for an entire library.

**Reply by C0nw0nk** [reply]
@lucas-0liveira 

I made a script for it.

https://github.com/C0nw0nk/DisableForcedSubtitles

Just run the script and paste the directory for example.

`Y:\TV Shows\Selling Sunset\`
OR
`Y:\TV Shows\`

You can also use `\\NETWORK-SHARE\MEDIA-SERVER\TV Shows`

If you want to remove extras like chapter markers etc these lines.
https://github.com/C0nw0nk/DisableForcedSubtitles/blob/main/DisableForceSubtitles.cmd#L18

You can run it via command line on a schedule with task scheduler.
Here is the command line for it.
```
:: Make script configurable via command line with arguements example
:: "C:\path\DisableForceSubtitles.cmd" "\\NAS\path" "remove_all_tags" "remove_all_chapter_markers" "check_for_sidecar" "pause_window" "wait_interval" "looping" 2^>nul
:: Working example
:: "C:\path\DisableForceSubtitles.cmd" "\\NAS\path" "1" "1" "0" "1" "120" "1" 2^>nul
```

If you use windows task scheduler to run it every 24 hours you can use it like this.
at the end all 0 means no pause no wait interval and no looping so the script will close then in 24 hours when your task scheduler runs again it will open it again.
```
"C:\path\DisableForceSubtitles.cmd" "\\NAS\path" "1" "1" "0" "0" "0" "0" 2^>nul
```



Enjoy!

**Comment by C0nw0nk** [comment]
@ptr727 I stumbled across this because it could be important.

So i run plexcleaner converts from mp4 to mkv etc you get your plexcleaner side car file to go with this.

I check the media its a h264 has subtitles no subtitle flags are forced.

Then i run --deinterlace to trigger handbrake and from what i can see plexcleaner through the deinterlace process does check the sidecar file but will then deinterlace (reconvert) the file and suddenly there are forced subtitle flags and subtitle codec has changed...

I do not know if that was intentional behavior did you know it does this even after reading the sidecar file ?

I stumbled across this thread where allot of other people have found handbrake will set a forced flag on the subtitle when the original subtitle never had one https://www.reddit.com/r/handbrake/comments/12ps814/subtitles_are_forced_but_not_set_that_way/

https://github.com/HandBrake/HandBrake/issues/4848

**Reply by ptr727** [reply]
`deinterlace` just runs HB to deinterlace, there is no conditional processing, and no working around HB subtitle conversion, that logic is only executed when using the `process` command. Sidecar files are just used to read the media attributes, it does nto change the processing behavior.

**Reply by C0nw0nk** [reply]
Ah i see i assumed plexcleaner.exe was doing pre handbrake checks.

What i will do is just run the script after deinterlace to resolve the behavior ASS subtitles is fine for me since i'm not running equipment from 2017 or earlier but those on older equipment will have to take it into consideration if they do not support ASS codecs.

**Reply by ptr727** [reply]
Readme: `These commands have no conditional logic and will process all specified media files.` ;)
